contracts/SnappBase.sol

```diff
@@ -109,23 +133,30 @@ contract SnappBase is Ownable {
     )
         public onlyOwner()
     {   
-        require(slot < depositSlot(), "Deposit slot must exist and be inactive");
-        require(depositHashes[slot].applied == false, "Deposits already processed");
-        require(depositHashes[slot].shaHash != bytes32(0), "Deposit slot is empty");
+        require(slot <= depositIndex(), "Requested deposit slot does not exist");
+
+        require(slot == 0 || deposits[slot-1].appliedAccountStateIndex != 0, "Must apply deposit slots in order!");
+        require(deposits[slot].appliedAccountStateIndex == 0, "Deposits already processed");
+        require(block.number > deposits[slot].creationBlock + 20, "Requested deposit slot is still active");
```

Comment:
@josojo This line here says that any deposit slot can be applied, even if nobody has deposited for a long time. Notice the same comparison with the block time + 20. This means that deposit slots become in active after 20 blocks, even if nobody deposits in the next one.